File: app/src/main/java/com/kotlin/android/adapters/ItemsAdapter.java
package com.kotlin.android.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * Name : ReceivingVerifyInvoicesAdapter
 * <br/> Created by 1694 on 21/8/17
 * <br/> Modified by 1694 on 21/8/17
 * <br/> Purpose : To show all started invoices in ReceivingVerifyFragment
 */
public class ItemsAdapter extends RecyclerView.Adapter<ItemsAdapter.ViewHolder> {
    private List<String> mList;

    public ItemsAdapter(List<String> aList)
    {
        this.mList = aList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
    {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(android.R.layout.simple_list_item_1, parent, false));
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position)
    {
        holder.bind(mList.get(position));
    }

    @Override
    public int getItemCount()
    {
        return mList == null ? 0 : mList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
/*        @BindView(R.id.tvVendorId)
        AppCompatTextView tvVendorId;

        @BindView(R.id.tvInvoiceNumber)
        AppCompatTextView tvInvoiceNumber;

        @BindView(R.id.tvInvoiceDate)
        AppCompatTextView tvInvoiceDate;

        @BindView(R.id.llItemView)
        LinearLayout llItemView;*/

        ViewHolder(View itemView)
        {
            super(itemView);
//            ButterKnife.bind(this, itemView);
        }

        public void bind(final String mItem)
        {
            try
            {
//                tvVendorId.setText(mItem);
                /*tvInvoiceNumber.setText(mItem.getInvoiceNumber());
                String invoiceDate = DateTimeUtils.getFormattedDate(mItem.getInvoiceDate());
                if (invoiceDate == null)
                {
                    tvInvoiceDate.setText(mItem.getInvoiceDate());
                } else
                {
                    tvInvoiceDate.setText(invoiceDate);
                }*/
            } catch (Exception e)
            {
                e.printStackTrace();
            }
        }
    }
}

File: app/src/main/java/com/kotlin/android/adapters/ItemsAdapter.kt
package com.kotlin.android.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.bvchopda.fragmentsdemo.R
import kotlinx.android.synthetic.main.layout_module_item.view.*

/**
 * Created by cygnet on 3/1/18.
 */
class ItemsAdapter(private val mList: List<String>?, private var onItemClickListener: OnItemClickListener) : RecyclerView.Adapter<ItemsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemsAdapter.ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.layout_module_item, parent, false))
    }

    override fun onBindViewHolder(holder: ItemsAdapter.ViewHolder, position: Int) {
        holder.bind(mList!![position])
        holder.itemView.setOnClickListener({
            onItemClickListener?.onItemClicked(holder.layoutPosition)
        })
    }

    override fun getItemCount(): Int {
        return mList?.size ?: 0
    }

    class ViewHolder
    (itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bind(mItem: String) {
            itemView.tvTitle.text = mItem
        }
    }

    interface OnItemClickListener {
        fun onItemClicked(index: Int)
    }

}